tests/unit/commands/publish/test_command.py

```diff
@@ -0,0 +1,147 @@
+"""Test sam publish CLI."""
+import json
+from unittest import TestCase
+from mock import patch, call, Mock
+
+from botocore.exceptions import ClientError
+
+from serverlessrepo.exceptions import ServerlessRepoError
+from serverlessrepo.publish import CREATE_APPLICATION, UPDATE_APPLICATION
+
+from samcli.commands.publish.command import do_cli as publish_cli
+from samcli.commands.local.cli_common.user_exceptions import SamTemplateNotFoundException
+from samcli.commands.exceptions import UserException
+
+
+class TestCli(TestCase):
+
+    def setUp(self):
+        self.template = "./template"
+        self.application_id = "arn:aws:serverlessrepo:us-east-1:123456789012:applications/hello"
+        self.ctx_mock = Mock()
+        self.ctx_mock.region = "us-east-1"
+        self.console_link = "Click the link below to view your application in AWS console:\n" \
+            "https://console.aws.amazon.com/serverlessrepo/home?region={}#/published-applications/{}"
+
+        path_patcher = patch('samcli.commands.publish.command.os.path')
```

Comment:
Can we follow what we have done in all of our testing to this point with adding annotations on the methods. Not a huge fan of the start and cleanup things that have to be done and it doesn't match the rest of the codebase.